airbyte-protocol/protocol-models/src/main/java/io/airbyte/protocol/models/CatalogHelpers.java

```diff
@@ -322,16 +325,25 @@ public static Set<StreamTransform> getCatalogDiff(final AirbyteCatalog oldCatalo
         .forEach(descriptor -> {
           final AirbyteStream streamOld = descriptorToStreamOld.get(descriptor);
           final AirbyteStream streamNew = descriptorToStreamNew.get(descriptor);
+
+          final List<ConfiguredAirbyteStream> streamList = configuredCatalog.getStreams().stream()
+              .filter(s -> Objects.equals(s.getStream().getNamespace(), descriptor.getNamespace())
+                  && s.getStream().getName().equals(descriptor.getName()))
+              .toList();
+
+          final ConfiguredAirbyteStream stream = streamList.get(0);
```

Comment:
These two statements could be combined into one statement which removes the need to have the `streamList` variable by replacing `toList()` with `findFirst()`
```
final var stream = configuredCatalog.getStreams().stream()
  .filter(s -> Objects.equals(s.getStream().getNamespace(), descriptor.getNamespace())
    && s.getStream().getName().equals(descriptor.getName()))
  .findFirst();
```

Additionally, as the `stream` variable is only referenced if `!streamOld.equals(streamNew)` this work would be moved to inside that `if` statement.  No reason to do this work if there is a chance it's going to be ignored.

The `findFirst()` call returns an `Optional` so you would need to check if you actually received a value.

Reply:
Also there is an issue with the current code, what if the `filter` call returns no results? The `streamList.get(0)` line will fail with an `IndexOutOfBoundsException`.